apps/api-extractor/src/extractor/Extractor.ts

```diff
@@ -164,11 +164,50 @@ export class Extractor {
    * @param options - IExtractor options.
    */
   public static processProjectFromConfigFile(jsonConfigFile: string, options?: IExtractorOptions): void {
-    const configObject: IExtractorConfig = JsonFile.loadAndValidate(jsonConfigFile, Extractor.jsonSchema);
+    const configObject: IExtractorConfig = this.loadConfigObject(jsonConfigFile);
     const extractor: Extractor = new Extractor(configObject, options);
     extractor.processProject();
   }
 
+  /**
+   * Loads the api extractor config file in Extractor Config object.
+   * @param jsonConfigFile - Path to api extractor json config file.
+   */
+  public static loadConfigObject(jsonConfigFile: string): IExtractorConfig {
+    // Set to keep track of config files which have been processed.
+    const pathSet: Set<string> = new Set<string>();
+    // Get absolute path of config file.
+    let currentConfigFilePath: string = path.resolve(jsonConfigFile);
+    pathSet.add(currentConfigFilePath);
+
+    let extractorConfig: IExtractorConfig = JsonFile.load(jsonConfigFile);
+
+    while (extractorConfig.extends) {
+      // Populate the api extractor config path defined in extends relative to current config path.
+      currentConfigFilePath = path.resolve(path.dirname(currentConfigFilePath), extractorConfig.extends);
```

Comment:
>extends [](start = 96, length = 7)

Can you add a bit of logic here that:

- checks if the path in `extends` starts with `.` or `/` and resolve relative to the `currentConfigFilePath` (or to the disk root)
- if it doesn't start with `.` or `/`, take a dependency on the `resolve` package and call `resolve.sync` to resolve the path as a package relative to `currentConfigFilePath`?

That way you can check off the package resolution feature as well.

Reply:
@sunilsurana  said he'll do that in a separate PR

---
In reply to: [236477994](https://github.com/Microsoft/web-build-tools/pull/957#discussion_r236477994) [](ancestors = 236477994)

Reply:
I have a doubt.
Lets say if path given is package/file.json then it should resolve to <currentconfigfiledir>/node_modules/package/file.json as per the module resolution requirement because path does not contain . or /
Now if file.json contain extends= ./file2.json then should it resolve to <currentconfigfiledir>/file2.json or <currentconfigfiledir>/node_modules/package/file2.json?

Reply:
Also I guess resolve package sync return relative path. (I'll check it out). If the path is relative or contain dots then we might need to wrap it with path.resolve to get absolute path to keep track of files which are processed in Set

Reply:
>Now if file.json contain extends= ./file2.json then should it resolve to **$currentconfigfiledir/file2.json** ~~or $currentconfigfiledir/node_modules/package/file2.json?~~

**This** is the correct behavior.

I think `resolve.sync` returns an absolute path.

Reply:
Fixed.